node_modules/oae-admin/manageuser/js/manageuser.js

```diff
@@ -156,15 +160,86 @@ define(['jquery', 'oae.core'], function($, oae) {
+         */
+        var deleteUser = function() {
+            // Lock the modal so it cannot be closed while deleting the user
+            $('#manageuser-modal', $rootel).modal('lock');
+            $('#manageuser-actions-form .modal-footer .manageuser-actions-deleteuser button').prop('disabled', true);
+
+            oae.api.admin.deleteUser(widgetData.userProfile.id, function(err) {
+                // When the delete request finishes re-enable the form
+                $('#manageuser-modal', $rootel).modal('unlock');
+                $('#manageuser-actions-form .modal-footer .manageuser-actions-deleteuser button').prop('disabled', false);
+
+                if (err) {
+                    // Show an error notification
+                    oae.api.util.notification(
+                        oae.api.i18n.translate('__MSG__USER_NOT_DELETED__', 'manageuser'),
+                        oae.api.i18n.translate('__MSG__COULD_NOT_DELETE_USER_NAME__', 'manageuser', {
+                            'displayName': oae.api.util.security().encodeForHTML(widgetData.userProfile.displayName)
+                        }),
+                        'error');
+                } else {
+                    // Success, reload the page
+                    document.location.reload()
+                }
+            });
+        };
+
+        /**
          * Set up the actions form
          */
         var setUpActionsForm = function() {
-            oae.api.util.template().render($('#manageuser-actions-template', $rootel), {
+            // Make the main actions panel the active one
+            showActionsPanel('main');
+
+            var data = {
                 'context': widgetData.context,
                 'userProfile': widgetData.userProfile
-            }, $('#manageuser-actions-form .modal-body', $rootel));
+            };
+
+            // Render the main panel body
+            oae.api.util.template().render($('#manageuser-actions-body-main-template', $rootel), data,
+                $('#manageuser-actions-form .modal-body .manageuser-actions-main', $rootel));
+
+            // Render the deleteuser panel body and footer
+            oae.api.util.template().render($('#manageuser-actions-body-deleteuser-template', $rootel), data,
+                $('#manageuser-actions-form .modal-body .manageuser-actions-deleteuser', $rootel));
+
+            oae.api.util.template().render($('#manageuser-actions-footer-deleteuser-template', $rootel), data,
+                $('#manageuser-actions-form .modal-footer .manageuser-actions-deleteuser', $rootel));
+
+            // Switch to deleteuser panel when user selects to delete the user
+            $rootel.on('click', '#manageuser-delete-user', function() {
+                showActionsPanel('deleteuser');
+            });
+
+            // Delete the user when the user accepts the deleteuser confirmation
+            $rootel.on('click', '#manageuser-actions-form .manageuser-deleteuser-confirm', deleteUser);
```

Comment:
I would prefer the confirm button and cancel button to have ids, as they will only be in the page once. In that case, there should also be no need to prefix with `#manageuser-actions-form`